src/app/questions-container/questions-container.component.ts

```diff
@@ -0,0 +1,116 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { HttpClientModule, HttpClient } from '@angular/common/http';
+import { CommonModule } from '@angular/common';
+import { HeaderComponent } from '../header/header.component';
+import { ScorePageComponent } from '../score-page/score-page.component';
+
+interface Question {
+  question: string;
+  options: string[];
+  answer: string;
+}
+
+interface Quiz {
+  title: string;
+  icon: string;
+  questions: Question[];
+}
+
+interface QuizData {
+  quizzes: Quiz[];
+}
+
+@Component({
+  selector: 'app-questions-container',
+  standalone: true,
+  imports: [
+    CommonModule,
+    HeaderComponent,
+    HttpClientModule,
+    ScorePageComponent,
+  ],
+  templateUrl: './questions-container.component.html',
+  styleUrls: ['./questions-container.component.css'],
+})
+export class QuestionsContainerComponent implements OnInit {
+  @Input() public subject = '';
+  public quizzes: Quiz[] = [];
+  @Input() public selectedQuiz: Quiz = {
+    title: '',
+    icon: '',
+    questions: [],
+  };
+  public currentQuestionIndex = 0;
+  public selectedAnswer: string | null = null;
+  public errorMessage: string | null = null;
+
+  public quizCompleted = false;
+  public correctAnswers = 0;
+
+  constructor(readonly http: HttpClient) {}
+
+  public ngOnInit() {
+    this.http.get<QuizData>('/data.json').subscribe((data) => {
```

Comment:
this should be in a service not here.
This means that create a service and move the http call to there.
also handle the quize in the filtering there